File: display/app/src/main/java/com/driveon/MainActivity.java
package com.driveon;

import android.app.Activity;
import android.os.Bundle;
import android.view.WindowManager;

public class MainActivity extends Activity {

    private FrameSurfaceView view;
    private TelemetryManager telemetry;
    private NetworkWorker worker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // 1. Inicializa Sensores e Tela
        telemetry = new TelemetryManager(this);
        view = new FrameSurfaceView(this);
        setContentView(view);

        // 2. Mantém tela ligada
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);

        // 3. Inicia Worker de Rede
        worker = new NetworkWorker(view, telemetry);
        worker.start();
    }

    @Override
    protected void onResume() {
        super.onResume();
        telemetry.start();
    }

    @Override
    protected void onPause() {
        super.onPause();
        telemetry.stop(); // Importante para salvar bateria se o app for para segundo plano
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        worker.shutdown();
    }
}

File: display/app/src/main/java/com/example/driveondisplay/MainActivity.kt
package com.example.driveondisplay

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.WindowManager

class MainActivity : Activity() {

    private lateinit var sensors: SensorClient
    private lateinit var view: GForceView
    private lateinit var sender: NetworkSender

    private val state = GForceState()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

//        NetworkClient.connectAsync()

//        sensors = SensorClient(this, state)
//        view = GForceView(this, state)
//        sender = NetworkSender(state)

        setContentView(FpsTestView(this))
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        hideSystemUI()
    }

    override fun onResume() {
        super.onResume()
//        sensors.start()
//        view.start()
//        sender.start()
    }

    override fun onPause() {
        super.onPause()
//        sensors.stop()
//        view.stop()
//        sender.stop()
    }

    private fun hideSystemUI() {
        window.decorView.systemUiVisibility =
            View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY or
                    View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or
                    View.SYSTEM_UI_FLAG_FULLSCREEN
    }
}
